object.go

```diff
@@ -548,6 +548,18 @@ func (o *baseObject) setOwnStr(name unistring.String, val Value, throw bool) boo
 			o.val.runtime.typeErrorResult(throw, "Cannot add property %s, object is not extensible", name)
 			return false
 		} else {
+			if o.val.runtime.shouldForceMemCheck {
+				memCtx := newMemUsageContextClone()
+				if memCtx != nil {
+					memUsage, err := val.MemUsage(memCtx)
+					if err != nil {
+						panic(err)
+					}
+					if memCtx.MemUsageLimitExceeded(memUsage) {
+						panic("memory limit exceeded")
```

Comment:
same q here about panicking with the func name.

Reply:
We don't have access to the function name here where we panic. I could try and see if we can pass it down from baas otherwise it might be cleaner to pick this up from baas when we catch this panic. Will leave this open until I have a definitive answer.

Reply:
In goja I def don't have access to the function name, I think we could potentially figure out the currently executing function within the App Services function but I'd rather investigate that later.